pkg/ibmvpc/vpc.go

```diff
@@ -278,21 +278,21 @@ func (nl *NaclLayer) StringDetailsRulesOfFilter(listRulesInFilter []vpcmodel.Rul
 	strListRulesInFilter := ""
 	for _, rulesInFilter := range listRulesInFilter {
 		nacl := nl.naclList[rulesInFilter.Filter]
-		header := getHeaderRulesType(networkACLStr+nacl.Name(), rulesInFilter.RulesFilterType) +
+		header := getHeaderRulesType(nl.Name()+" "+nacl.Name(), rulesInFilter.RulesFilterType) +
```

Comment:
you mean `nl.Kind()` instead of `nl.Name()`, and the same for `sg`. In which case the functions `Name()` of `SecurityGroupLayer` and `NaclLayer` can be restored to what they were - and return the empty string. This is what you had in mind?
However, that will result in SecurityGroupLayer and NaclLayer instead of security group and network acl. So I guess the answer is "No" :-)

Reply:
I meant to be using `nacl.Kind()`, not `nl.Kind()`.
The `kind` is available from a specific filter resource, not from the entity that represents the entire layer.
Thus if we could use that `kind` instead of adding a "name" for a layer entity (to actually indicate the kind of the filter objects that it holds), that could be more consistent.

Reply:
The current implementation is a bit confusing, as both nacl's kind and nacl-layer's name are returning the same string that indicates the filter type

Reply:
The former returns `NaclLayer`  and the latter returns "network ACL"  which is not the same, and also not used for the same purposes. Added explanation in documentation, hope it is cleared

Reply:
no, nacl's Kind returns "NACL", nacl-layer's kind returns "NACL Layer" , and now also nacl-layer's name returns "NACL"

Reply:
I was referring to *nacl layer's* Kind(). You are referring to *nacl* Kind().
1. The context I am generating the relevant printing and its header is the layer's context. Using for the header Kind() of a specific nacl (sg) is possible but not natural and will make the code messier.
2. Having the additional Name() for the nacl layer gives us more flexibility - we can indeed choose the best wording for the printing which could be (and will probably be) different than its "Kind()".

Reply:
> I was referring to nacl layer's Kind(). You are referring to nacl Kind().

right, but it seemed that both are returning the same result, which is the type name per single filter object on its layer.

> Having the additional Name() for the nacl layer gives us more flexibility

Maybe this should not be called `Name()` , but rather `FilterObjectKind()` or something like that.. The `Name` method is part of the interface `VPCResourceIntf` , and in all other places it's a name of a resource from the config object.. so it is inconsistent to have this method return another string result in a different context. 
It can be a new method specific to the `FilterTrafficResource` interface.

Reply:
But the layer does not have a "personal" name... So the name of the layer is in fact "network ACL" (or whatever we decide to call it).
Anyways, I'll add another function as you suggest, given that Name() is used only for "personal" names.